### PR TITLE
add: enable webhook patching with flag

### LINKED ISSUE #6184: Certificate generation should respect the state of the webhooks
### Report

Currently, we support generating the certificates in the operator using cert-controller if users don't want to use cert-manager. The problem is that the operator tries to patch the webhook resource always but the webhook is optional, so when the users are using operator managed certificates but disable the webhooks, KEDA operator fails patching the webhook resources (because it doesn't exist)

We should include a flag in the operator to include the webhook patching as part of the loop or exclude it

### Expected Behavior

.

### Actual Behavior

.

### Steps to Reproduce the Problem

1.
2.
3.

### Logs from KEDA operator

```
example
```

### KEDA Version

None

### Kubernetes Version

None

### Platform

None

### Scaler Details

_No response_

### Anything else?

.

### PR DESCRIPTION
**Pull request by KrishnaSindhur**
Includes a flag in the operator to the webhook patching 

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs/pull/1509)) - 
- [x] A PR is opened to update our Helm chart on ([repo](https://github.com/kedacore/charts/pull/715)) - 
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #6184

### CODE REVIEW

File: pkg/certificates/certificate_manager.go [hunk 1]
```diff
@@ -50,19 +50,27 @@ type CertManager struct {
...
-		{
-			Name: cm.APIServiceName,
-			Type: rotator.APIService,
-		},
+	var rotatorHooks []rotator.WebhookInfo
+
+	rotatorHooks = append(rotatorHooks, rotator.WebhookInfo{
+		Name: cm.APIServiceName,
+		Type: rotator.APIService,
+	})
```
Comment (wozniakjan) [inline pkg/certificates/certificate_manager.go:63]: nit: saving a few CPU cycles :)
```suggestion
	rotatorHooks := []rotator.WebhookInfo{
        {
		    Name: cm.APIServiceName,
		    Type: rotator.APIService,
	    },
	}
```

**Review by wozniakjan — APPROVED** [review]
lgtm, thank you!

**Review by zroubalik — COMMENTED** [review]
Looking good, 

Could you please update docs and helm as well?

**Review by JorTurFer — APPROVED** [review]
This is awesome! Could you open a PR updating docs as well?

**Pull request merged**